tests/test_installables.py

```
@@ -0,0 +1,169 @@
+
+    @pytest.mark.parametrize(
+        ("init_submodules", "stdout", "expected_command"),
+        [
+            (True, "-0123456789abcdef path/to/submodule\n", ["git", "submodule", "update", "--init", "--recursive"]),
+            (False, " 0123456789abcdef path/to/submodule\n", ["git", "submodule", "deinit", "--all", "--force"]),
+            (False, "+0123456789abcdef path/to/submodule\n", ["git", "submodule", "deinit", "--all", "--force"]),
+            (False, "U0123456789abcdef path/to/submodule\n", ["git", "submodule", "deinit", "--all", "--force"]),
+        ],
+    )
+    def test_ensure_submodules_state_reconciles(
+        self, git: GitRepo, init_submodules: bool, stdout: str, expected_command: list[str]
```

Ruff (0.15.10)

[warning] Boolean-typed positional argument in function definition

(FBT001)